sklearn/inspection/_h_statistic.py

```diff
@@ -0,0 +1,260 @@
+"""Friedman and Popescu's H-Statistic"""
+
+import itertools
+
+import numpy as np
+
+from ..base import is_classifier, is_regressor
+from ..utils import Bunch, _get_column_indices, _safe_assign, _safe_indexing
+from ..utils.random import sample_without_replacement
+from ..utils.validation import _check_sample_weight, check_is_fitted
+
+
+def _calculate_pd_brute_fast(estimator, X, feature_indices, grid, sample_weight=None):
+    """Fast version of _calculate_partial_dependence_brute()
+
+    Returns np.array of size (n_grid, ) or (n_ngrid, output_dimension).
+    """
+
+    if is_regressor(estimator):
+        pred_fun = estimator.predict
+    elif is_classifier(estimator) and hasattr(estimator, "predict_proba"):
+        pred_fun = estimator.predict_proba
+    else:
+        raise ValueError("The estimator has no predict or predict_proba method.")
```

Comment:
The error message should distinguish between regressors and classifiers. For regressors, `predict` is missing, for classifiers `predict_proba`.

Reply:
I have now the slightly lengthy but explicit:

```
    if is_regressor(estimator):
        if hasattr(estimator, "predict"):
            pred_fun = estimator.predict
        else:
            raise ValueError("The regressor has no predict() method.")
    elif is_classifier(estimator):
        if hasattr(estimator, "predict_proba"):
            pred_fun = estimator.predict_proba
        else:
            raise ValueError("The classifier has no predict_proba() method.")
```

Reply:
That's fine. But we can move that validation to `h_statistic` such that it happens only once.

Reply:
Doing this only once would be much neater. But then we would need to pass "pred_fun" instead of "estimator" to  `_calculate_pd_over_data()` and `_calculate_pd_brute_fast()`. I don't know if this is okay?

Reply:
Sure, passing a callable as function argument, in particular internally, is totally fine in Python.